cluster/fpindexcluster/manager.py

```
@@ -611,12 +671,32 @@ async def publish_delete_index(self, index_name: str) -> None:
             # Publish the operation to the index stream
             await self._publish_operation(index_name, DeleteIndexOperation())
 
+    async def publish_update(self, index_name: str, changes: list[Change], metadata: dict = None) -> None:
+        """
+        Publish an update operation to the index stream.
+        Validates that the index exists and is active before publishing.
+        """
+        # Check if index exists and is active
+        status = await self.get_index_status(index_name)
+        if not status.status.active:
+            raise InconsistentIndexState(f"Index '{index_name}' is not active (cannot perform updates)")
+        
+        # Ensure stream exists for this index (should already exist if index is active)
+        await self._ensure_stream_exists(index_name)
```

Skip this check, just let the operation fail if it does not exist.